unity-renderer/Assets/Rendering/Culling/CullingObjectsTracker.cs

```diff
@@ -1,144 +1,198 @@
+            detectedSkinnedRenderers = null;
+            usingAnimations = null;
+            PoolManager.i.OnGet -= MarkDirty;
+            Rendereable.OnRendereableAdded -= OnRendereableAdded;
+        }
+
+        /// <summary> Sets the dirty flag to true to make PopulateRenderersList retrieve all the scene objects on its next call. </summary>
+        public void MarkDirty() => dirty = true;
+
+        /// <summary> Returns true if dirty. </summary>
+        public bool IsDirty() => dirty;
+
+        /// <summary> Returns the renderers list. </summary>
+        public ICollection<Renderer> GetRenderers() => usingRenderers;
+
+        /// <summary> Returns the skinned renderers list. </summary>
+        public ICollection<SkinnedMeshRenderer> GetSkinnedRenderers() => usingSkinnedRenderers;
+
+        /// <summary> Returns the animations list. </summary>
+        public Animation[] GetAnimations() => usingAnimations;
+
+
+        private void ForceCalculateRenderers()
+        {
+            ForceCleanRendereables();
+            
+            usingRenderers = new List<Renderer>();
+            foreach (Rendereable rendereable in rendereables)
             {
-                if (renderer == null)
+                if (!rendereable.container.activeInHierarchy)
                     continue;
 
-                if (amount >= CullingControllerSettings.MAX_POPULATING_ELEMENTS_PER_FRAME)
+                foreach (Renderer renderer in rendereable.renderers)
                 {
-                    yield return null;
-                    amount = 0;
-                }
-                amount++;
+                    if (renderer == null || !renderer.gameObject.activeInHierarchy)
+                        continue;
 
-                if ((((1 << renderer.gameObject.layer) & ignoredLayersMask) == 0)
-                    && !(renderer is ParticleSystemRenderer))
-                {
-                    if (renderer is SkinnedMeshRenderer)
-                        skinnedRenderersList.Add((SkinnedMeshRenderer)renderer);
-                    else
-                        renderersList.Add(renderer);
+                    if (((1 << renderer.gameObject.layer) & ignoredLayersMask) == 0)
+                        usingRenderers.Add(renderer);
                 }
             }
-            renderers = renderersList;
-            skinnedRenderers = skinnedRenderersList;
-
-            yield return null;
+        }
 
-            animations = Object.FindObjectsOfType<Animation>();
+        private void ForceCalculateSkinnedRenderers()
+        {
+            usingSkinnedRenderers = new List<SkinnedMeshRenderer>(detectedSkinnedRenderers);
+            foreach (SkinnedMeshRenderer skinnedRenderer in detectedSkinnedRenderers)
+            {
+                if (skinnedRenderer == null)
+                    continue;
 
-            dirty = false;
+                if (((1 << skinnedRenderer.gameObject.layer) & ignoredLayersMask) == 0)
```

Comment:
Try to avoid repetitions: you have layer checking in two functions: consider moving this piece into a separate method